src/Datadog.Trace.ClrProfiler.Managed/Emit/ModuleLookup.cs

```diff
@@ -0,0 +1,46 @@
+using System;
+using System.Collections.Concurrent;
+using System.Reflection;
+using Datadog.Trace.Logging;
+
+namespace Datadog.Trace.ClrProfiler.Emit
+{
+    internal static class ModuleLookup
+    {
+        private static readonly ILog Log = LogProvider.GetLogger(typeof(ModuleLookup));
+
+        private static ConcurrentDictionary<Guid, Module> _modules = new ConcurrentDictionary<Guid, Module>();
+
+        public static Module Get(Guid moduleVersionId)
+        {
+            if (!_modules.ContainsKey(moduleVersionId))
+            {
+                PopulateModules();
+            }
+
+            if (_modules.ContainsKey(moduleVersionId))
+            {
+                return _modules[moduleVersionId];
+            }
+
+            return null;
+        }
+
+        private static void PopulateModules()
+        {
+            var assemblies = AppDomain.CurrentDomain.GetAssemblies();
+            foreach (var assembly in assemblies)
+            {
+                foreach (var module in assembly.Modules)
+                {
+                    if (_modules.ContainsKey(module.ModuleVersionId))
```

Comment:
No need to call `ContainsKey()` when using `TryAdd()`. If the key already exists, it returns `false` (which we can ignore in this case).